packages/hoprd/crates/hoprd-misc/src/config.rs

```diff
@@ -49,30 +61,44 @@ fn validate_api_auth(token: &Auth) -> Result<(), ValidationError> {
 #[derive(Debug, Default, Serialize, Deserialize, Validate, Clone, PartialEq)]
 pub struct Host {
     #[validate(custom = "validate_ipv4_address")]
-    pub ip: String,
+    pub ip: Option<String>,
```

Comment:
This is not backwards friendly to wasm. A better approach would be to use enum and type, e.g:

```rust
pub enum Host {
  IPv4(String),
  Domain(String)
}
```

Make this member conditionally public (if not wasm) and add checkers (get_ip -> Option<String>) with internally tranforming the enum to wasm recognized value.

Reply:
As agreed will be done in config PR.